### PR TITLE
cln: Boltz auto-send

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**
- Bug fix: Hide claim tx id and routing fee with zero conf reverse swap
- Added auto send feature for swap in

## Summary by CodeRabbit

- **New Features**
	- Introduced an "Auto Send" option for swaps, enhancing user control over transaction processes.
	- Display of QR codes for transactions, improving user experience in swap status updates.
- **Enhancements**
	- Optimized data fetching for swap operations, ensuring a smoother user experience.
	- Improved user interface styling for better visibility and accessibility, particularly in dark mode.
- **Refactor**
	- Streamlined swap-related components for efficiency and maintainability.
- **Documentation**
	- Added new models and services documentation to support the Boltz swap functionality.

### CODE REVIEW

File: src/app/shared/models/boltzModels.ts [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+  node: string;
+  network: string;
+  nodePubkey: string;
+  autoSwapStatus: string;
+  blockHeights: BlockHeight[];
+  symbol: string;
+  lndPubkey: string;
+  blockHeight: number;
+  pendingSwaps: string[];
+  pendingReverseSwaps: string[];
```
Comment (jackstar12) [inline src/app/shared/models/boltzModels.ts:17]: These fields are deprecated

File: src/app/shared/models/boltzModels.ts [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+  value: number;
+}
+
+export interface BoltzInfo {
+  version: string;
+  node: string;
+  network: string;
+  nodePubkey: string;
+  autoSwapStatus: string;
+  blockHeights: BlockHeight[];
```
Comment (jackstar12) [inline src/app/shared/models/boltzModels.ts:12]: The actual response is a map:
```json
  "blockHeights": {
    "BTC": 171,
    "L-BTC": 150
  },
```
But im not sure why doc generator says its an array:
https://docs.boltz.exchange/v/boltz-client/grpc#getinforesponse

**Comment by saubyk** [comment]
@coderabbitai review

File: src/app/shared/models/boltzModels.ts (hunk 1 above)
Comment (ShahanaFarooqui) [reply src/app/shared/models/boltzModels.ts]: Removed.

File: src/app/shared/models/boltzModels.ts (hunk 2 above)
Comment (ShahanaFarooqui) [reply src/app/shared/models/boltzModels.ts]: Updated.

File: server/controllers/shared/boltz.ts [hunk 3]
```diff
@@ -89,7 +89,8 @@ export const createSwap = (req, res, next) => {
   }
   options.url = options.url + '/v1/createswap';
   options.body = { amount: req.body.amount };
-  if (req.body.address !== '') { options.body.address = req.body.address; }
+  if (req.body.autoSend) { options.body.auto_send = req.body.autoSend; }
+  if (req.body.address && req.body.address !== '') { options.body.address = req.body.address; }
```
[the remark this answers is not in the document]
Comment (ShahanaFarooqui) [reply server/controllers/shared/boltz.ts]: Will change all controllers code accordingly in a separate PR.

**Review by jackstar12 — CHANGES_REQUESTED** [review]
The address is still shown if the swap is being paid internally, which makes the user think the swap still needs to be paid. Only show the lockup transaction in this case, which is included in the [response](https://docs.boltz.exchange/v/boltz-client/grpc#createswapresponse):

**Comment by ShahanaFarooqui** [comment]
> The address is still shown if the swap is being paid internally, which makes the user think the swap still needs to be paid. Only show the lockup transaction in this case, which is included in the [response](https://docs.boltz.exchange/v/boltz-client/grpc#createswapresponse):

Updated the code accordingly with [commit](https://github.com/Ride-The-Lightning/RTL/commit/9db447d31cd4ea31cc2784a115001efcb397e3e2) .

**Pull request merged**